pandas/tests/groupby/test_counting.py

```diff
@@ -195,3 +196,28 @@ def test_ngroup_respects_groupby_order(self):
                                 g.ngroup())
             assert_series_equal(Series(df['group_index'].values),
                                 g.cumcount())
+
+    def test_count_with_datetimelike(self):
+        # test for #13393, where DataframeGroupBy.count() fails
+        # when counting a datetimelike column.
+
+        expected = DataFrame({'y': [2, 1]}, index=['a', 'b'])
+        expected.index.name = "x"
+
+        def _test(datetimelike):
+            df = DataFrame({'x': ['a', 'a', 'b'], 'y': datetimelike})
+            res = df.groupby('x').count()
+            assert_frame_equal(expected, res)
+
+        d1 = [Timestamp('2016-05-07 20:09:25+00:00'),
+              Timestamp('2016-05-07 20:09:29+00:00'),
+              Timestamp('2016-05-07 20:09:29+00:00')]
+        _test(d1)
+        d2 = [Timedelta(x, unit="h") for x in range(1, 4)]
+        _test(d2)
```

Comment:
you need to change Timedelta and Period to work here

Reply:
I am not sure I understand your request.

Reply:
the examples for timedelta and period are unique and won’t group to the same as the other examples 

just change to individually construct them like you did for time stamps

Reply:
but we are not grouping by the timestamps, it doesn't matter if they are unique or not; the original issue fails regardless.